### PR TITLE
Tekton demo prowjob update

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Changes proposed in this pull request:

- Added updated version of tekton demo prowjob.

**Related issue(s)**
See #6798

### CODE REVIEW

File: docs/examples/tekton-demo.yaml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+              configMap:
+                name: kaniko-build-config
+            - name: signify-secret
+              secret:
+                secretName: signify-dev-secret
+            - name: image-registry-credentials
+              secret:
+                secretName: sa-kyma-push-images
+          # PipelineRun provide values of params used by the pipeline.
+          # Prow implicitly and automatically set values of params according to the list of job environment variables.
```
Comment (IwonaLanger) [inline docs/examples/tekton-demo.yaml:45]:
```suggestion
          # Prow implicitly and automatically sets params values according to the list of job environment variables.
```

File: docs/examples/tekton-demo.yaml [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+          params:
+            - name: name
+              value: test-infra/test-image
+            - name: config
+              value: /config/kaniko-build-config.yaml
+            - name: dockerfile
+              value: development/secrets-rotator/cloud-run/rotate-service-account/Dockerfile
+          # The following resources are not used by the pipeline, but are required by the ProwJob syntax validator.
+          # They are not used by the pipeline because the pipeline uses environment variables to get extra_refs to clone the repo in a task.
+          # This is a workaround for deprecated resource object in Tekton Pipeline api version v1beta1.
```
Comment (IwonaLanger) [inline docs/examples/tekton-demo.yaml:55]:
```suggestion
          # This is a workaround for a deprecated resource object in Tekton Pipeline API version v1beta1.
```

File: docs/examples/tekton-demo.yaml [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+          resources:
+            - name: extra-ref
+              resourceRef:
+                name: PROW_EXTRA_GIT_REF_0
+          # Inline pipeline specification.
+          # The pipelineSpec does not contain implementation details of workspaces and params because they are propagated from the PipelineRun.
+          pipelineSpec:
+            # The following resources are not used by the pipeline, but are required by the ProwJob syntax validator.
+            # They are not used by the pipeline because the pipeline uses environment variables to get extra_refs to clone the repo in a task.
+            # This is a workaround for deprecated resource object in Tekton Pipeline api version v1beta1.
```
Comment (IwonaLanger) [inline docs/examples/tekton-demo.yaml:65]:
```suggestion
            # This is a workaround for a deprecated resource object in Tekton Pipeline API version v1beta1.
```

File: docs/examples/tekton-demo.yaml [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+            resources:
+              - name: extra-ref
+                type: git
+            description: "Tekton Demo"
+            # TaskRuns executed by teh pipieline. TaskRuns provide values and implementation details required by Tasks executed by the pipeline.
```
Comment (IwonaLanger) [inline docs/examples/tekton-demo.yaml:70]:
```suggestion
            # TaskRuns executed by the pipeline. TaskRuns provide values and implementation details required by Tasks executed by the pipeline.
```

File: docs/examples/tekton-demo.yaml [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+            # TaskRuns provide values and implementations explicitly or binding them to values and implementations provided by the pipeline.
```
Comment (IwonaLanger) [inline docs/examples/tekton-demo.yaml:71]:
```suggestion
            # TaskRuns provide values and implementations explicitly or bind them to values and implementations provided by the pipeline.
```

File: docs/examples/tekton-demo.yaml [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+            tasks:
+              # Task name.
+              - name: clone-refs
+                # A name of a Task to execute. A Task specification must exist in the pipelines cluster.
```
Comment (IwonaLanger) [inline docs/examples/tekton-demo.yaml:75]:
```suggestion
                # The name of the Task to execute. The Task specification must exist in the pipeline cluster.
```

File: docs/examples/tekton-demo.yaml [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+                # Bindings of workspaces used by the Task to workspaces provided by the pipeline.
+                workspaces:
+                  - name: repo
+                    workspace: repo
+                # Bindings of params used by the Task to params provided by the pipeline.
+                params:
+                  - name: JOB_SPEC
+                    value: $(params.JOB_SPEC)
+              # TaskRun with inline task specification.
+              # TskRuns with inline tasks can be mixed with TaskRuns referencing existing tasks by name in the same pipeline.
```
Comment (IwonaLanger) [inline docs/examples/tekton-demo.yaml:87]:
```suggestion
              # TaskRuns with inline tasks can be mixed with TaskRuns referencing existing tasks by name in the same pipeline.
```

File: docs/examples/tekton-demo.yaml [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+                    workspace: repo
+                taskSpec:
+                  workspaces:
+                    - description: The workspace where sources to build an image are stored.
+                      mountPath: /home/prow/go
+                      name: repo
+                  steps:
+                    - name: config
+                      image: ubuntu
+                      # Using script is a way to run inlined script. This doesn't have to be a shell script.
```
Comment (IwonaLanger) [inline docs/examples/tekton-demo.yaml:102]:
```suggestion
                      # Using a script is a way to run an inline script. This doesn't have to be a shell script.
```

File: docs/examples/tekton-demo.yaml [hunk 9]
```diff
@@ -0,0 +1,184 @@
+presubmits:
+  kyma-project/test-infra:
+    - name: tekton-demo
+      # Default Prow decoration must be disabled because the pipeline uses its own decoration.
+      decorate: false
+      # Prowjobs using Tekton pipelines must use the tekton-pipeline agent.
+      agent: tekton-pipeline
+      # Tekton has its own dedicated cluster to run pipelines. Use tekton cluster to for pipeline prowjobs.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply docs/examples/tekton-demo.yaml]: A tekton is a name of a cluster not the name of a tool. Left it as it is. Rest of suggestion applied.

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**